scalaz-test/src/main/scala/ztest/tests/ExhaustiveSuite.scala

```diff
@@ -0,0 +1,28 @@
+package scalaz.test
+package tests
+
+import scala._
+
+import scalaz.data.IList
+import scalaz.Monad
+import scalaz.Scalaz._
+import property._
+
+final class ExhaustiveSuite extends PureSuite {
+  def doTests[G[_]: Monad](harness: Harness[() => ?, G]): G[Unit] = {
```

Comment:
Is `Monad` constraint necessary? How would one do any sort of introspection on the tests, say for SBT?

Reply:
Oh no, you said _the forbidden word_...